core/domain/state_domain.py

```diff
@@ -4818,6 +4877,10 @@ def update_interaction_customization_args(
         Raises:
             Exception. The customization arguments are not unique.
         """
+        # Here we use assert so that the customization_arg_dict will become
```

Comment:
If this is for mutability, then why not MutableMapping?

In general, "avoid mypy complaining" is not a sufficient reason on its own.

Reply:
>If this is for mutability, then why not MutableMapping?

MutableMapping does not allow the assignment of dict subtypes, and I think this is already concluded while discussion using this link(https://github.com/python/mypy/issues/8136) as a ref.

> In general, "avoid mypy complaining" is not a sufficient reason on its own.

More explicit reason is added!